functions/ocr/app/detect.go

```diff
@@ -0,0 +1,129 @@
+// Copyright 2018, Google, LLC.
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//    http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ocr
+
+// [START functions_ocr_detect]
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"io"
+	"io/ioutil"
+
+	pubsub "cloud.google.com/go/pubsub/apiv1"
+	"cloud.google.com/go/translate"
+	vision "cloud.google.com/go/vision/apiv1"
+	"golang.org/x/text/language"
+	pb "google.golang.org/genproto/googleapis/cloud/vision/v1"
+	pubsubpb "google.golang.org/genproto/googleapis/pubsub/v1"
+)
+
+// detectText detects the text in an image using the Google Vision API.
+func detectText(w io.Writer, projectID, bucketName, fileName string) error {
+	// bucketName := "ocr-image-bucket123"
+	// fileName := "menu.jpg"
+	fmt.Fprintf(w, "Looking for text in image %v", fileName)
+	ctx := context.Background()
+	visionClient, err := vision.NewImageAnnotatorClient(ctx)
+	if err != nil {
+		return fmt.Errorf("vision.NewImageAnnotatorClient: %v", err)
+	}
+	maxResults := 1
+	annotations, err := visionClient.DetectTexts(ctx,
+		&pb.Image{
```

Comment:
Create a separate variable for the image and change this function call to a single line.